lib/lens/worker.rb

```diff
@@ -0,0 +1,133 @@
+module Lens
+  class Worker
+    # A queue which enforces a maximum size.
+    class Queue < ::Queue
+      attr_reader :max_size
+
+      def initialize(max_size)
+        @max_size = max_size
+        super()
+      end
+
+      def push(obj)
+        super unless size >= max_size
+      end
+    end
+
+    class << self
+      def instance
+        @instance
+      end
+
+      def running?
+        !instance.nil?
+      end
+
+      def start(config)
+        return instance if running?
+        @instance = new(config)
+      end
+
+      def stop(force = false)
+        @instance.send(force ? :shutdown! : :shutdown)
+      end
+    end
+
+    SHUTDOWN = :__lens_worker_shutdown!
+
+    attr_reader :config, :queue, :pid, :mutex, :thread
+
+    def initialize(config)
+      @config = config
+      @queue = Queue.new(100)
+      @mutex = Mutex.new
+      @shutdown = false
+      start
+    end
+
+    def push(obj)
+      if start
+        queue.push(obj)
+      end
+    end
+
+    def start
+      mutex.synchronize do
+        return false if @shutdown
+        return true if thread && thread.alive?
+
+        @pid = Process.pid
+        @thread = Thread.new { run }
+      end
+
+      true
+    end
+
+    def run
+      begin
+        loop do
+          case msg = queue.pop
+          when SHUTDOWN
+            break
+          else
+            process(msg)
+          end
+        end
+      end
+    rescue Exception => e
```

Comment:
Avoid rescuing the `Exception` class. Perhaps you meant to rescue `StandardError`?<br>Useless assignment to variable - `e`.